src/project/controller/person-collection.ts

```diff
@@ -17,22 +15,12 @@ class ProjectPersonCollection extends Controller {
     const params = {
       role: ctx.request.body.role,
       name: ctx.request.body.name,
-      href: ctx.request.body.href
+      href: ctx.request.body.href,
     };
-
-    const person = await projectService.addPersonToProject(params);
     const projectId = +ctx.params.projectId;
-    const project = await projectService.findById(projectId);
-
-    if(!person.principalUri){
-      throw new NotFound(`The principal_uri is missing on Person: ${person.id}.`);
-    }
+    const origin = ctx.request.origin;
 
-    await addUserPrivilege(
-      person.principalUri,
-      params.role,
-      new URL(project.href, ctx.request.origin),
-    );
+    await projectService.addPersonToProject(params, projectId, origin);
```

Comment:
So this object is now starting to look really weird. What are params, what aren't params? Why are some things in params? Decide on a path, either a parameters object, or do every argument separately.

Also generally avoid passing something like a `projectId`, you always want to pass the full object.